gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -447,6 +450,72 @@ gst_tensor_filter_init (GstTensorFilter * self)
   gst_tensors_config_init (&self->out_config);
 }
 
+/**
+ * @brief Clear and reset data.
+ */
+static void
+gst_tensor_filter_reset (GstTensorFilter * self)
+{
+  GstTensorFilterProperties *prop;
+  guint i;
+
+  prop = &self->prop;
+
+  gst_tensor_filter_close_fw (self);
+
+  if (prop->fwname) {
+    g_free ((char *) prop->fwname);
+    prop->fwname = NULL;
+  }
+
+  if (prop->model_file) {
+    g_free ((char *) prop->model_file);
+    prop->model_file = NULL;
+  }
+
+  if (prop->custom_properties) {
+    g_free ((char *) prop->custom_properties);
+    prop->custom_properties = NULL;
+  }
+
+  for (i = 0; i < prop->input_meta.num_tensors; i++) {
+    if (prop->input_meta.info[i].name) {
+      g_free (prop->input_meta.info[i].name);
+    }
+  }
+
+  prop->input_configured = FALSE;
+  gst_tensors_info_init (&prop->input_meta);
+
+  for (i = 0; i < prop->output_meta.num_tensors; i++) {
+    if (prop->output_meta.info[i].name) {
+      g_free (prop->output_meta.info[i].name);
+    }
+  }
+
+  prop->output_configured = FALSE;
+  gst_tensors_info_init (&prop->output_meta);
```

Comment:
How about using macro or inline function to avoid code duplication between 481-488 and 490-497. The only difference between these two block is input_meta/output_meta.

Reply:
it's applied :)